frontends/python/futil/futil_ast.py

```diff
@@ -0,0 +1,432 @@
+from __future__ import annotations  # Used for circular dependencies.
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union
+from futil.futil_utils import block
+
+
+@dataclass
+class Emittable:
+    def doc(self) -> str:
+        assert 0, f'`doc` not implemented for {type(self).__name__}'
+
+    def emit(self):
+        print(self.doc())
+
+
+### Program ###
+
+@dataclass
+class Import(Emittable):
+    filename: str
+
+    def __init__(self, filename: str):
```

Comment:
I thought `dataclass` automatically generates the `__init__` method for you: https://docs.python.org/3/library/dataclasses.html

Reply:
Nice. Fixed.